@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -121,6 +121,27 @@ export enum PrePlacementResult {
     Count
 }
 
+/**
+ * Defines possible text placement relative to anchor.
+ */
+interface AnchorPlacement {
+    h: HorizontalAlignment;
+    v: VerticalAlignment;
+}
+
+/**
+ * @hidden
+ * Possible placement scenarios in clock-wise order.
+ *
+ * TODO: This array should be parsed from the style/theme definition of the technique.
+ */
+const possibleAnchorPlacementsCW = [
```

Comment:
What about {Center,Above}, {Right,Center} ... ? Don't we need those relative positions for China?

I would expect to have here 8 combinations, all except {Center,Center}. If a label has centered placement, let's assume so far it doesn't want alternative placements. That way we avoid testing alternative placements for labels without icons (they are centered in our styles).

Reply:
Good point of course! The problem is that with hard-coded anchors placement we would need to test 9 scenarios (8 on corners, and center-center), this is way too much I guess.

> That way we avoid testing alternative placements for labels without icons (they are centered in our styles).
Good catch I think we may use it as a pre-mature optimization which at least limits the computations, of course finally the best is to have it configured via theme .json.

Reply:
1. Do you agree that 8 anchors is too much, may we stay with fixed 4 placements + eventually current one if not existent on placements list?
2. Done - centered labels are no longer processed with multi-anchor algorithm.

Reply:
Done. We have two arrays for alternative placements:
- with centered positions (if that is the base one - in theme),
- with corner position (if theme has such layout).